heidrun/getty.rb

```diff
@@ -0,0 +1,134 @@
+LANGUAGE_MAP = {
+  'multiple languages' => 'mul'
+}
+
+DATA_PROVIDER_MAP = {
+  'GETTY_ROSETTA' => 'Getty Research Institute',
+  'GETTY_OCP' => 'Getty Research Institute'
+}
+
+build_url = lambda do |p|
+  source = p.root[Krikri::PrimoParser.record('control', 'sourceid').join('/')]
+           .first.value
+  id = p.root[Krikri::PrimoParser.record('control', 'recordid').join('/')]
+       .first.value
+
+  base_url = 'http://primo.getty.edu/primo_library/libweb/action/dlDisplay.do'
+
+  case source
+  when 'GETTY_ROSETTA'
+    "#{base_url}?vid=GRI&afterPDS=true&institution=01GRI&docId=#{id}"
+  when 'GETTY_OCP'
+    "#{base_url}?vid=GRI-OCP&afterPDS=true&institution=01GRI&docId=#{id}"
+  end
+end
+
+lookup_language = lambda do |i|
+  val = i.value.downcase
+  LANGUAGE_MAP.fetch(val, val)
+end
+
+Krikri::Mapper.define(:getty,
+                      parser: Krikri::PrimoParser) do
+  provider class: DPLA::MAP::Agent do
+    uri 'http://dp.la/api/contributor/getty'
+    label 'J. Paul Getty Trust'
+  end
+
+  dataProvider class: DPLA::MAP::Agent,
+               each: record.field(*Krikri::PrimoParser.record('control',
+                                                              'sourceid'))
+                      .map { |i| DATA_PROVIDER_MAP.fetch(i.value, i.value) }
+                      .first_value,
+               as: :provider do
+    label provider
+  end
+
+  isShownAt class: DPLA::MAP::WebResource do
+    uri build_url
+  end
+
+  preview class: DPLA::MAP::WebResource do
+    uri record.field('sear:LINKS', 'sear:thumbnail')
+  end
+
+  sourceResource class: DPLA::MAP::SourceResource do
+    title record.fields(Krikri::PrimoParser.display('title'),
+                        Krikri::PrimoParser.display('lds03'))
+           .map { |v| v.value.split(';').map(&:strip) }.flatten
+
+    # TODO: The original (python) getty mapping seemed to use lds34 and lds43
+    # here.  Should we?
+    collection class: DPLA::MAP::Collection,
+               each: record.fields(Krikri::PrimoParser.display('lds31'),
+                                   Krikri::PrimoParser.display('lds32')),
+               as: :coll do
+      title coll
+    end
+
+    contributor class: DPLA::MAP::Agent,
+                each: record.field(*Krikri::PrimoParser.display('contributor'))
+                  .map { |i| i.value.split(';').map(&:strip) }.flatten,
+                as: :contrib do
+      providedLabel contrib
+    end
+
+    creator class: DPLA::MAP::Agent,
+            each: record.field(*Krikri::PrimoParser.display('creator')),
+            as: :creator do
+      providedLabel creator
+    end
+
+    # TODO: Enrichment needed here to remove trailing punctuation and set
```

Comment:
I'm not sure what it would require to break out begin and end dates from their 'creationdate' field, but I'm guessing that it isn't necessary.  I assume that they can both be set to the same value (it being a single creation date), maybe even by our existing timespan-splitting enrichment (`timespan_split.rb`).  There are at least three providers for whom we do not assign any begin or end date in the mapping:  SCDL, DLG, and UW.